ITL-1 Implement Page header

EmanuelRusuu opened the pull request:
Feature: ITL-1 Implement Page header

EmanuelRusuu committed:
ITL-7 Create the header component

- created header component

EmanuelRusuu committed:
ITL-8 Add the title

- added "To do list" title

EmanuelRusuu committed:
ITL-9 Add an "add button"

- added the "add todo" button (included the icon from the figma design)
- changes to comply to the "Vue style guide"

EmanuelRusuu committed:
ITL-10 Create the event that triggers the todo creation functionality

- created an add button that adds tasks when the user clicks it
- solved conflicts for 4 files complying with ITL-10

roli-wg commented on todoapp/src/components/Header.vue:
```diff
@@ -1,16 +1,63 @@
 
-const emit = defineEmits(['deleteTasks'])
+const emit = defineEmits<{
+  addTodo: any
```
Please don't use any type you can type this emit like:
`(e: 'addToto', todo: string): void`

EmanuelRusuu committed:
Solved git changes request

- complied with vue style guide for type-based emits
- got rid of "any" type where needed
- replicated figma design

roli-wg commented on todoapp/src/components/Header.vue:
```diff
@@ -1,16 +1,71 @@
 
-function deleteTasks() {
-  emit('deleteTasks')
+const inputValue = computed({
```
You don't need to have this inputValue in the header component (you just have a title and a button :))

roli-wg commented on todoapp/src/components/Header.vue:
```diff
@@ -1,16 +1,71 @@
+})
+
+function handleAdd() {
+  emit('addTodo', inputValue.value)
```
You don't need to send this inputValue I think it's enough to just emit the addTodo action without any parameter

roli-wg commented on todoapp/src/components/TodoForm.vue:
```diff
@@ -5,8 +5,8 @@ const props = defineProps<{
-  addTodo: any
-  'update:modelValue': any
+  (e: 'addTodo', todo: string): void
+  (e: 'update:modelValue', value: any): void
```
Please don't use any type if possible

roli-wg commented on todoapp/src/components/Todo.vue:
```diff
@@ -0,0 +1,23 @@
+  todos: TextIdString[]
+}>()
+const emits = defineEmits<{
+  (e: 'removeTodo', todo: TextIdString): void
```
If you has and ID into todo object you don't have to send back the whole todo for delete action can be find and delete by id or even simple since you have an array of todo's you don't need that id and the edit and delete can be done base on the index

EmanuelRusuu committed:
Solved git changes request 2

For the first request:
- I removed the "inputValue" from the Header.vue component, because it was not needed anymore for the ITL-1.

For the second request:
- I removed the parameter "inputValue.value", from Header.vue and also from TodoForm.vue, because sending it to the parents was unnecessary.

For the third request:
- I removed all "any" types I could find, replacing them with proper ones.

For the fourth request:
- You suggested removing the "id" property from every todo in todos, and so I did. I used the index that comes with the array, and changed the removeTodo function to take "index" as parameter.

Other changes made:
- better styling;
- removed bits of unused code;
- imported the font-family used in Figma design;
- changed some code to comply with the VueStyleGuide.

EmanuelRusuu committed:
Solved git changes no-request 3

- removed unnecessary prop (v-model="inputContent") sent to Header component in TodoList.vue (ln. 38)
